src/Microsoft.AspNet.Razor/TagHelpers/TagHelperAttributeDescriptorComparer.cs

```diff
@@ -31,7 +31,8 @@ protected TagHelperAttributeDescriptorComparer()
         /// <remarks>
         /// Determines equality based on <see cref="TagHelperAttributeDescriptor.IsIndexer"/>,
         /// <see cref="TagHelperAttributeDescriptor.Name"/>, <see cref="TagHelperAttributeDescriptor.PropertyName"/>,
-        /// and <see cref="TagHelperAttributeDescriptor.TypeName"/>. Ignores
+        /// <see cref="TagHelperAttributeDescriptor.TypeName"/>, and
+        /// <see cref="TagHelperAttributeDescriptor.UsageDescriptor"/>. Ignores
```

Comment:
All that needs to change in this file is the comments. Add `UsageDescriptor` to the list of ignored properties because it (like `IsStringProperty`) can be inferred directly from `TypeName`.